content/tutorials/guides/how-to-get-pim-category-tree.md

```diff
@@ -0,0 +1,325 @@
+        'base_uri' => $pimUrl,
+        'headers' => ['Authorization' => 'Bearer ' . $appToken],
+    ]);
+}
+```
+
+```javascript [activate:NodeJS]
+
+// Install the node-fetch library by following the official documentation:
+// https://www.npmjs.com/package/node-fetch
+import fetch from 'node-fetch';
+
+const pimUrl = 'https://url-of-your-pim.com';
+const accessToken = 'your_app_token'; // Token provided during oAuth steps
+
+// Set your client for querying Akeneo API as follows
+async function get(url, accessToken) {
+  return await fetch(url, {
+    headers: {
+      'Authorization': `Bearer ${accessToken}`
+    }
+  });
+}
+```
+
+### 1. Get the PIM structure by fetching a channel from API
+
+Workflow:
+
+![synchronisation steps](/img/getting-started/synchronize-pim-products/step-3-steps-schema.svg)
+
+Collect flat category tree from PIM API:
+
+```php [activate:PHP]
+
+function getCategories(): void
+{
+    $client = buildApiClient();
+
+    $rootCategoryCode = 'master';
+    $maxItems = 100;
+
+    $nextUrl = sprintf(
+        '/api/rest/v1/categories'
+        . '?with_position=true'
+        . '&search={"parent":[{"operator":"=","value":"%s"}]}'
+        . '&limit=%s',
+        $rootCategoryCode,
+        $maxItems
+    );
+
+    $categoryPages = [];
+
+    do {
+        $response = $client->get($nextUrl);
+        $data = json_decode($response->getBody()->getContents(), true);
+        $categoryPages[] = $data['_embedded']['items'];
+
+        $nextUrl = $data['_links']['next']['href'] ?? null;
+    } while (
+        $nextUrl
+    );
+
+		// Adds root category to complete the list.
+    $categoryPages[][] = getRootCategory($rootCategoryCode);
+
+    $categories = array_merge(...$categoryPages);
+
+    saveCategories($categories);
+}
+
+function getRootCategory(string $code): array
+{
+    $client = buildApiClient();
+
+    $response = $client->get('/api/rest/v1/categories/' . $code);
+    $rootCategory = json_decode($response->getBody()->getContents(), true);
+
+    if ($rootCategory['parent'] !== null) {
+        throw new \InvalidArgumentException(
+            sprintf('Category %s is not root.', $code)
+        );
+    }
+
+    return $rootCategory;
+}
+```
+
+```javascript [activate:NodeJS]
+
+async function fetchCategories() {
+  const rootCategoryCode = 'master';
+  const maxItems = 100;
+
+  let nextUrl = `${pimUrl}/api/rest/v1/categories`
+    + '?with_position=true'
+    + `&search={"parent":[{"operator":"=","value":"${rootCategoryCode}"}]}`
+    + `&limit=${maxItems}`
+  ;
+
+  const categories = [];
+
+  do {
+    const response = await get(nextUrl);
+    const data = await response.json();
+    const newCategories = data._embedded.items;
+    categories.push(...newCategories);
+
+    nextUrl = data._links?.next?.href;
+  } while (nextUrl)
+
+  // Adds root category to complete the list.
+  const rootCategory = await getRootCategory(rootCategoryCode);
+  categories.push(rootCategory);
+
+  saveCategories(categories);
```

Comment:
add comments: the user doesn't know what is this function

Reply:
js-guided-tutorials: done